### PR TITLE
3771/ Fixed long category names go over the button on Homepage

### PR DESCRIPTION
**Pull request by hoahoang311**
Make button height fit content and text wrapped inside the button

Related issue(s):

- Fixes https://github.com/scandipwa/scandipwa/issues/3771

Problem:

- Text overflow the category button on the Home page

In this PR:

- Make button height fit content and text wrapped inside the button

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1. Button/text is displayed behind the button if screen width ±940px 
Safari:
Screenshot 2021-11-30 at 08 03 05
Chrome (Win):
Capture
_______________________
2. Looks like indent/padding is missing between button border and text on Windows and Android if compare with iOS and Mac
Windows:
Mac:
Screenshot 2021-11-30 at 08 14 40
Windows:
Screenshot 2021-11-30 081601

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Fixed
3. iOS: Text is cropped if the word does not fit on one line 
Screenshot 2021-12-02 at 12 56 58

4. Android: Text is displayed behind the button if the word does not fit on one line
Screenshot 2021-12-02 at 12 55 40

**Comment by aleksandrakorolova** [comment]
3. Fixed
4. Fixed
5. Button is displayed behind the button in big screen resolution 
Screenshot 2021-12-06 at 14 20 26

Display Resolution - 2560x1440 pixels
Screen Size - 25 inches

**Comment by aleksandrakorolova** [comment]
5. Fixed

File: packages/scandipwa/src/component/ProductAttributeValue/ProductAttributeValue.component.js [hunk 1]
```diff
@@ -279,7 +279,7 @@ export class ProductAttributeValue extends PureComponent {
               block="ProductAttributeValue"
               elem="SubLabel"
             >
-                { `(${subLabel})` }
+                { subLabel ? `(${subLabel})` : undefined }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductAttributeValue/ProductAttributeValue.component.js:282]: isn't it fixed in this PR?
https://github.com/scandipwa/scandipwa/pull/3838/files

File: packages/scandipwa/src/component/ProductAttributeValue/ProductAttributeValue.component.js [hunk 2]
```diff
@@ -307,15 +307,15 @@ export class ProductAttributeValue extends PureComponent {
...
+        const { isFormattedAsText, isSelected, isProductCountVisible } = this.props;
         const isSwatch = label;
 
         if (isFormattedAsText) {
             return label || value || __('N/A');
         }
 
         if (!isSwatch) {
-            return this.renderDropdown(value, count);
+            return this.renderDropdown(value, isProductCountVisible ? count : null);
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductAttributeValue/ProductAttributeValue.component.js:318]: Please check in renderDropdown function

File: packages/scandipwa/src/style/cms/block/homepage-category-preview.scss [hunk 3]
```diff
@@ -74,4 +75,27 @@
...
+        }
+
+        @include mobile {
+            max-width: 80vw;
+        }
+
+        * {
+            overflow-wrap: break-word;
+            word-wrap: break-word;
+        }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/style/cms/block/homepage-category-preview.scss:99]: To everything? Maybe we can restring it a bit for specific elements?

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Comment by carinadues** [comment]
Hey, @hoahoang311 can you please fix this merge conflict? Then we'll merge the PR :)

**Comment by hoahoang311** [comment]
Conflict resolved!

**Pull request merged**